### ISSUE TITLE
About Pre-Trained Model.

### ISSUE THREAD

**Issue by HojaMuerta** [sunlicai/MAE-DFER#8]
Thanks for your great work,

I wonder if the pre-training process is the same as MAE-DFER or just not different from Video-MAE? In other words, does your pre-trained model include the LGI-Former?

**Comment by sunlicai** [comment]
Hi! Thanks for your attention to our work!

The pre-trained model include the LGI-Former. It use LGI-Former as the encoder.

Hope this answer your question.

**Comment by HojaMuerta** [comment]
> Hi! Thanks for your attention to our work!
> 
> The pre-trained model include the LGI-Former. It use LGI-Former as the encoder.
> 
> Hope this answer your question.


Thank you for your response. I am currently attempting to pre-train my own model, but I have encountered several issues with the process  including LGI-Former. I am unsure if you can provide the code for pre-training.

**Comment by sunlicai** [comment]
I am quite busy these days due to the deadline of current work. I will upload the pre-training code after the deadline and it shall be available in this month. Stay tuned, Thanks!

**Comment by HojaMuerta** [comment]
Thanks a lot!!!

**Comment by HojaMuerta** [comment]
Sorry to bother you again, but I would like to know the specific details of the application of LGI-Former in the pre-training process. Is only the part that is not obscured or all the pictures sent into the model training? In this process, does Local global attention apply to the pictures before or after the mask? How is the corresponding patch sent to the model?

**Comment by sunlicai** [comment]
Good question! 

During pre-training, only 10% visible (i.e., unmasked) tokens will be processed by LGI-Former, which is different from that in fine-tuning. Thus, the local global attention is applied to the tokens after tube masking.  

Note that, to ensure that all local regions have an equal number of visible tokens, we apply the same masking ratio to each local region during tube masking.

**Comment by HojaMuerta** [comment]
If I use a 2D picture with a size of 224*224 and a patch size of 16, can it be understood that the picture is first divided into 14*14 patches, and then a fixed proportion mask is applied to each patch, so that the visible part of each Local region is the same? Then LGI-Former is applied to each masked local region. Here, LGI-Former focuses on the unmask part or the whole part of each patch. Or does LGI-Former process the unmask part of each patch before concat feeding it into the model? Thank you very much!

**Comment by HojaMuerta** [comment]
I mean do I apply LGI-Former to the unmask part of each patch separately and then add up the results into the model or do I connect the unmask part of each patch together and then use LGI-Former? If it is the former, how is the value of lg_region_size set?RSVP!

**Comment by sunlicai** [comment]
If I understood correctly, you should connect the unmask part of each patch together and then use LGI-Former.

**Comment by HojaMuerta** [comment]
Okay, thanks for taking the time to respond, and I'll run the experiment.

**Comment by HojaMuerta** [comment]
I have completed the experiment, but compared with the classic static picture MAE, the training time of a single epoch is significantly increased. Is it because of the addition of LGI-Former? Is this normal?

**Comment by sunlicai** [comment]
I did not observe significant epoch time increase. It might be caused by the implementation.

**Comment by HojaMuerta** [comment]
Ok, there may be a mistake in my implementation, I hope you upload your pre-training code at your leisure.

**Comment by youcaiSUN** [comment]
Hi @HojaMuerta, we've already uploaded the pre-training code.

**Comment by HojaMuerta** [comment]
> Hi @HojaMuerta, we've already uploaded the pre-training code. 

So appreciate!
